From 03068ada4373ad31d453f8bc8b061b4020b32358 Mon Sep 17 00:00:00 2001
From: Jon Miranda <jonmiranda@google.com>
Date: Thu, 1 Apr 2021 13:14:23 -0400
Subject: [PATCH] Set FLAG_IMMUTABLE flag on FirstScreenBroadcast
 PendingIntent.

Bug: 183927137
Test: restore phone and ensure broadcast is sent
Change-Id: I8d8e38a1aa6bdf13879d460cfa84cabe6c6bb1f2
(cherry picked from commit b62fba0d0199d719ac63761846e09b4b28e1dc2c)
---
 src/com/android/launcher3/model/FirstScreenBroadcast.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/model/FirstScreenBroadcast.java b/src/com/android/launcher3/model/FirstScreenBroadcast.java
index 5112304fb..f9ed2c78b 100644
--- a/src/com/android/launcher3/model/FirstScreenBroadcast.java
+++ b/src/com/android/launcher3/model/FirstScreenBroadcast.java
@@ -15,6 +15,8 @@
  */
 package com.android.launcher3.model;
 
+import static android.app.PendingIntent.FLAG_IMMUTABLE;
+import static android.app.PendingIntent.FLAG_ONE_SHOT;
 import static android.os.Process.myUserHandle;
 
 import android.app.PendingIntent;
@@ -145,7 +147,7 @@ public class FirstScreenBroadcast {
                 .putStringArrayListExtra(HOTSEAT_ITEM_EXTRA, new ArrayList<>(hotseatItems))
                 .putStringArrayListExtra(WIDGET_ITEM_EXTRA, new ArrayList<>(widgetItems))
                 .putExtra(VERIFICATION_TOKEN_EXTRA, PendingIntent.getActivity(context, 0,
-                        new Intent(), PendingIntent.FLAG_ONE_SHOT)));
+                        new Intent(), FLAG_ONE_SHOT | FLAG_IMMUTABLE)));
     }
 
     private static String getPackageName(ItemInfo info) {
-- 
2.17.1

